### ISSUE TITLE
[MacOS] RC7 Bugs

### ISSUE THREAD

**Issue by ExperiBass** [LMMS/lmms#4718]
Why is RC7 so wonky? I've encountered many bugs on MacOS.

**Screen refresh issues** 

- Whenever I open a project, phantom windows of the Song editor, B&B, FX/Mixer, and Controller Rack appear, covering the windows. the problem will only be fixed by moving a window to make the scroll bars appear. (image below)

- Dual position cursors may appear when LMMS comes back into focus. the second cursor can be "erased" by moving the mouse cursor over the second cursor. ("Dual cursor")
[LMMS RC7 Bugs.zip](https://github.com/LMMS/lmms/files/2627416/LMMS.RC7.Bugs.zip)

- whenever I change the color of a B/B track, it will not change until a) the position marker move over it or b) the Song Editor is resized. ("BB color bug 1.png" & "BB color bug 2.png")

**Export hang issue**

- Whenever I export, the percentage gets stuck at 99% and LMMS will hang until forced to quit. the terminal log and crash report are in the zip file below. ("LMMS render terminal output" & "exporting bug")

**Other**

- VSTs are not supported, but there's is a VST folder and the option to set a VST folder. what gives? the terminal output seems to suggest I need a `Carla.app`. ("Terminal output")

**Edited:** 1/29 by @tresf

**Comment by PhysSong** [comment]
Could you upload a project with the hang on export? I think I can investigate the hang.

**Comment by GingkathFox** [comment]
[GingkathFox - Melodic Noise.mmpz.zip](https://github.com/LMMS/lmms/files/2627619/GingkathFox.-.Melodic.Noise.mmpz.zip)
@PhysSong

**Comment by GingkathFox** [comment]
![screen shot 2018-11-29 at 12 10 57 pm](https://user-images.githubusercontent.com/33764485/49239167-03f5fd00-f3d0-11e8-9181-4a780110a67e.png)
Phantom window bug pic

**Comment by musikBear** [comment]
@GingkathFox -Could you please edit the title from 
> RC7 Bugs

To
"MAC specific RC7 Bugs"

**Comment by SecondFlight** [comment]
The macOS label should probably removed if these are reproduced anywhere else.

**Comment by tresf** [comment]
Some Macs have screen refresh issues.  They've been documented before.  If you're running the old build (e.g. macOS 10.8 build), see if the newer build helps.  Also, please try compiling from source using the latest Qt5 version from Homebrew and see if they exist.

In regards to the vst folder, that should be hidden by the precompiler, but no one ever coded it.  Help welcome.  Many Linux distros build without VST support, so that's not Mac specific.

To summarize this bug report...

1. Screen refresh issues
2. Crash on export, but no details to reproduce

I've been using macOS as my main workstation for quite a while.  LMMS works great (Mojave on one machine, El Capitan on the other).

😄

**Comment by tresf** [comment]
Here's the previously reported refresh issues: https://github.com/LMMS/lmms/issues/2504

**Comment by GingkathFox** [comment]
@tresf tried building from source, same screen refresh issue. and im already using 10.13+ build.

**Comment by tresf** [comment]
@GingkathFox 5.11.2 claims to address a very similar issue: https://bugreports.qt.io/browse/QTBUG-68740

**Comment by tresf** [comment]
> VSTs are not supported, but there's is a VST folder and the option to set a VST folder. what gives? the terminal output seems to suggest I need a Carla.app. ("Terminal output")

Please move this to a dedicated bug report.

**Comment by tresf** [comment]
* VST is tracked elsewhere: #2393 #3975 (duplicate)
* Screen tearing issues occur on other platforms (e.g. #4577) and have been closed by upstream bug reports (please reopen a separate issue if these still exist). 
* Export hang sounds like the only valid bug, but too much has changed in 6 years to track it here.  Closing.  Feel free to reference this in a new bug report if any of these issues can be reproduced on master/nightly
